lib/resources/datasets.js

```diff
@@ -7,11 +7,33 @@
 // including this file, may be copied, modified, propagated, or distributed
 // except according to the terms contained in the LICENSE file.
 const { getOrNotFound } = require('../util/promise');
+const { Form } = require('../model/frames');
+const { success } = require('../util/http');
 
 module.exports = (service, endpoint) => {
   service.get('/projects/:id/datasets', endpoint(({ Projects, Datasets }, { auth, params, queryOptions }) =>
     Projects.getById(params.id, queryOptions)
       .then(getOrNotFound)
       .then((project) => auth.canOrReject('project.read', project)) // TODO change the verb
       .then(() => Datasets.getAllByProjectId(params.id))));
+
+
+  service.post('/projects/:projectId/forms/:id/draft/attachments/:name/link-dataset', endpoint(({ Datasets, FormAttachments, Forms }, { auth, params }) =>
```

Comment:
I believe this would be our first major deviation from REST since `link-dataset` is an action (so more RPC-like). I haven't given this deep thought yet but I wonder whether there might be a version with `PATCH` and a new property on attachments that would stay within REST. Maybe it's a combination of a new `type` version `dataset` and `datasetid`?

I see this is a draft so don't mean to derail your chain of thought, I'm excited to see this addition and the route jumped out at me as I took a peek!

Reply:
I like the idea of using `PATCH` too (with `type: dataset`), since it's about doing something to the attachment "slot".

`DELETE` could unlink if it is a dataset.

Reply:
Talked to @matthew-white a bit about this and he pointed that `type` has multiple other meanings in/near this context:
- content type of the blob (sent in the existing attachment `POST` [request](https://odkcentral.docs.apiary.io/#reference/forms/draft-form/uploading-a-draft-form-attachment) and then served back with the blob data)
- type of the attachment as figured out from the form XML (parsed around [here](https://github.com/getodk/central-backend/blob/master/lib/data/schema.js#L326) somewhere?) e.g. `image`, `audio`, `video`, and the only one that is valid for datasets: `file`

So another alternative could be something like `PATCH` with `{ dataset: true }` and avoid the word "type".

Reply:
implemented PATCH